apps/fluent-tester/src/E2E/Switch/specs/Switch.spec.win.ts

```diff
@@ -0,0 +1,103 @@
+  });
+});
+
+describe('Switch Accessibility Testing', () => {
+  /* Scrolls and waits for the Switch to be visible on the Test Page */
+  beforeEach(() => {
+    SwitchPageObject.scrollToTestElement();
+    SwitchPageObject.waitForPrimaryElementDisplayed(PAGE_TIMEOUT);
+  });
+
+  it('Switch - Validate accessibilityRole is correct', () => {
+    expect(SwitchPageObject.getAccessibilityRole()).toEqual(BUTTON_A11Y_ROLE);
+    expect(SwitchPageObject.didAssertPopup()).toBeFalsy(SwitchPageObject.ERRORMESSAGE_ASSERT);
+  });
+});
+
+describe('Switch Functional Testing', () => {
+  /* Scrolls and waits for the Switch to be visible on the Test Page */
+  beforeEach(() => {
+    SwitchPageObject.scrollToTestElement();
+    SwitchPageObject.waitForPrimaryElementDisplayed(PAGE_TIMEOUT);
+  });
+
+  it("Click on a Switch -> Validate it toggles correctly AND calls the user's onChange", () => {
+    /* Validate the Switch is initially toggled OFF */
+    expect(SwitchPageObject.isSwitchChecked()).toBeFalsy();
+
+    /* Click on the Switch to toggle on */
+    SwitchPageObject.clickComponent();
+    SwitchPageObject.waitForSwitchChecked(PAGE_TIMEOUT);
+
+    expect(SwitchPageObject.didOnChangeCallbackFire()).toBeTruthy();
+
+    /* Validate the Switch is toggled ON */
+    expect(SwitchPageObject.isSwitchChecked()).toBeTruthy();
+
+    SwitchPageObject.clickComponent();
+
+    /* Validate the Switch is toggled OFF */
+    expect(SwitchPageObject.isSwitchChecked()).toBeFalsy();
+    expect(SwitchPageObject.didAssertPopup()).toBeFalsy(SwitchPageObject.ERRORMESSAGE_ASSERT);
+  });
+
+  it('Click the "Enter" on a Switch and verify it toggles', () => {
+    /* Validate the Switch is initially toggled OFF */
+    expect(SwitchPageObject.isSwitchChecked()).toBeFalsy();
+
+    /* Presses the "Enter" to select the Switch */
+    SwitchPageObject.sendKey(SwitchComponentSelector.PrimaryComponent, Keys.Enter);
+    SwitchPageObject.waitForSwitchChecked(PAGE_TIMEOUT);
+
+    /* Validate the Switch is toggled ON */
+    expect(SwitchPageObject.isSwitchChecked()).toBeTruthy();
+    expect(SwitchPageObject.didAssertPopup()).toBeFalsy(SwitchPageObject.ERRORMESSAGE_ASSERT);
+
+    SwitchPageObject.sendKey(SwitchComponentSelector.PrimaryComponent, Keys.Enter);
+
+    /* Validate the Switch is toggled OFF */
+    expect(SwitchPageObject.isSwitchChecked()).toBeFalsy();
+    expect(SwitchPageObject.didAssertPopup()).toBeFalsy(SwitchPageObject.ERRORMESSAGE_ASSERT);
+  });
+
+  it('Click the "Spacebar" on a Switch and verify it toggles', () => {
+    /* Validate the Switch is initially toggled OFF */
+    expect(SwitchPageObject.isSwitchChecked()).toBeFalsy();
+
+    /* Presses the "space bar" to select the Switch */
+    SwitchPageObject.sendKey(SwitchComponentSelector.PrimaryComponent, Keys.Spacebar);
+    SwitchPageObject.waitForSwitchChecked(PAGE_TIMEOUT);
+
```

Comment:
`expect(SwitchPageObject.didOnChangeCallbackFire()).toBeTruthy();`

Reply:
Added that in